### DISCUSSION TITLE
Trying to add F4 Discovery Board but having an issue.

### DISCUSSION THREAD

**Discussion by EricB-Arch** [grblHAL/core#543] [Q&A] [answered]
Hi, I'm working on adding the F4 Disco. 
1) Having an issue where in building Release the board might hard fault or or I get a USB not recognized in windows. 

If I run a debug via VS Code/Platform IO and break on Line 2899 "return hal.version == 10;" in driver.c for some reason it will inititialize fine. Probably something I did.

Here is my commit, be great if someone would take a look at it for any glaring errors.
https://github.com/EricB-Arch/STM32F4xx/commit/7362eeac5e1d2f16994744669df43f9abd16b0b5#diff-4446afd728a4f34cbcddc306a9cb6be845d1a61c216076a295683bcc9c106724

2) Is it possible to set number of and ganged axis in platormio.ini? Haven't figured that out.

edit: Ok, I see  I have a problem with the pin assignments for M4 or M5, N_AXIS=4 is working!

Thanks for a very cool project and any help.

**Comment by EricB-Arch** [comment]
For ganging I see the answer is probably in motor_pins but first some sleep.

**Comment by terjeio** [comment]
1. Hard to tell what is causing this since I do not have a board to test with.
2. Most #defines can be overridden from platformio.ini (with -D ...) - ganged axes are configured by [these](https://github.com/grblHAL/STM32F4xx/blob/2cfeccf4053071133d90ec02e0faef7706b92d94/Inc/my_machine.h#L129-L141).

**Reply by EricB-Arch** [reply]
Thank you, I did go back and better understand the #defines and ran into to some issues in my board map. Mostly sorted now.

Still having the release vs. debug issue, first I am assuming it's something with my configuration. I'd guess there is some race going on where something I did is breaking USB or pin conflict maybe. Will keep at it!

**Reply by EricB-Arch** [reply, marked as answer]
Defined uart pins and it fixed the release issue.
